server/api/product.js

```diff
@@ -0,0 +1,63 @@
+const router = require('express').Router()
+const { Product, Review, Category} = require('../db/models')
+
+router.param('productId', function (req, res, next, productId) {
+  Product.findById(productId)
+    .then(product => {
+      req.product = product;
+      next()
+    })
+})
+
+router.get('/', function (req, res, next) {
+  Product.findAll({
+    include: [{
+      model: Category
+    }]
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+
+router.get('/:productId', function (req, res, next) {
+  Product.findAll({
+    where: { productid: req.params.productId },
+    include: [{ model: Review }]
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+
+router.delete('/:productId', function (req, res, next) {
+  Product.destroy({ where: { id: req.params.productId } })
+    .then(() => res.sendStatus(204))
+    .catch(next)
+})
+
+router.put('/:productId', function (req, res, next) {
+  Product.findById(req.params.id)
+    .then(function (product) {
```

Comment:
You used es6 earlier on, so here you could also use it by doing

.then(product => product.update(req.body)
.then(updatedProduct => res.send(updatedProduct)
.catch(next)